### PR TITLE
feat(3015991255): close the modal if click on the blurry part

### PR DESCRIPTION
**Pull request by galuzan-starkware**
### Description of the Changes

add a blurry background `.modal` CSS class and added property: `exitable` on the `ModalReducer`, which will close the modal if it's true with `hideModal` and prevent closing the modal when the user is clicking on the children of the blurry part by using `stopPropagation.`

### Checklist

- [ ] Changes have been done against master branch, and PR does not conflict
- [ ] New unit / functional tests have been added (whenever applicable)
- [ ] Test are passing in local environment
- [ ] Docs have been updated
- [ ] PR title is follow the [Conventional Commits](https://www.conventionalcommits.org/) convention: `<type>[optional scope]: <description>`, e.g: `fix: prevent racing of requests`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starkgate-frontend/259)

### CODE REVIEW

File: src/components/UI/Modal/Modal/Modal.js [hunk 1]
```diff
@@ -31,17 +31,23 @@ export const Modal = ({
   show,
   type = ModalType.INFO,
   size = ModalSize.MEDIUM,
+  children,
```
Comment (dan-ziv) [inline src/components/UI/Modal/Modal/Modal.js:34]: Should be place last

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/components/UI/Modal/Modal/Modal.js (hunk 1 above)
Comment (dan-ziv) [reply src/components/UI/Modal/Modal/Modal.js:34]: Again, children should be the last prop.
Second, don't pass the `hideModal` as a prop from outside, you can use the `ModalWrapper` to pass it (we're using `useHideModal` inside the  ModalWrapper)

**Review by dan-ziv — CHANGES_REQUESTED** [review]

**Comment by dan-ziv** [comment]
@galuzan-starkware The description should describe the way of your solution and not what you already wrote in the title.

**Comment by dan-ziv** [comment]
@galuzan-starkware Please check no bugs in Firefox.

File: src/components/UI/Modal/Modal/Modal.js [hunk 2]
```diff
@@ -31,17 +31,23 @@ export const Modal = ({
...
+  hideModal
 }) => {
   const {width} = size;
 
   return show
     ? createPortal(
-        <div className={toClasses(styles.modal, styles[type])}>
+        <div
+          className={toClasses(styles.modal, styles[type])}
+          onClick={() => exitable && hideModal()}
```
Comment (dan-ziv) [inline src/components/UI/Modal/Modal/Modal.js:45]: Export to named function (anonymous functions as event handlers are bad practice).

File: src/components/UI/Modal/Modal/Modal.js [hunk 3]
```diff
@@ -31,17 +31,23 @@ export const Modal = ({
...
+        >
           <div
             className={toClasses(styles.container, styles[type])}
             style={{width, maxWidth: width, ...containerStyle}}
+            onClick={e => e.stopPropagation()}
```
Comment (dan-ziv) [inline src/components/UI/Modal/Modal/Modal.js:50]: Export to named function

**Review by dan-ziv — CHANGES_REQUESTED** [review]

**Comment by galuzan-starkware** [comment]
> @galuzan-starkware The description should describe the way of your solution and not what you already wrote in the title.

done

**Comment by galuzan-starkware** [comment]
> @galuzan-starkware Please check no bugs in Firefox.

There are no bugs relative to my changes, but I notice that if there is no MetaMask extension, then the app is crush

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**